pkg/controllers/management/usercontrollers/controller.go

```diff
@@ -82,13 +82,8 @@ func (c *ClusterLifecycleCleanup) Remove(obj *v3.Cluster) (runtime.Object, error
 		err = c.cleanupImportedCluster(obj)
 	}
 	if err != nil {
-		apiError, ok := err.(*httperror.APIError)
-		// If it's not an API error give it back
-		if !ok {
-			return nil, err
-		}
-		// If it's anything but clusterUnavailable give it back
-		if apiError.Code != httperror.ClusterUnavailable {
+		// If it's anything but cluster agent disconnected give it back
+		if !strings.Contains(err.Error(), dialer.ErrAgentDisconnected.Error()) {
```

Comment:
I think this is a more idiomatic approach.
```suggestion
		if errors.Is(err, dialer.ErrAgentDisconnected) {
```